tests/test_celery.py

```diff
@@ -256,3 +257,13 @@ def test_method_signatures(celery_app, async_task, method, _value):
     if method == "run":
         return
     assert inspect.signature(getattr(async_task.__class__, method)) == inspect.signature(getattr(Task, method))
+
+
+def test__get_backend_returns_DisabledBackground_object_when_result_backend_is_set_to_None(notify_celery):
+    notify_celery.conf.update({"result_backend": None})
+    assert isinstance(notify_celery._get_backend(), DisabledBackend)
```

Comment:
This would give you some confidence that your code is doing what you want (not trying to discover any backends):
```suggestion
    mock_by_url = mocker.patch("celery.app.backends.by_url")
    mock_by_name = mocker.patch("celery.app.backends.by_name")
    assert isinstance(notify_celery._get_backend(), DisabledBackend)
    assert mock_by_name.called is mock_by_url.called is False
```

Otherwise this test will pass with or without your changes.